packages/hop-node/src/watchers/xDomainMessageRelayWatcher.ts

```diff
@@ -71,29 +83,52 @@ class xDomainMessageRelayWatcher extends BaseWatcher {
       })
     }
 
+
+    const messengerWrapperContract: MessengerWrapperContract = contracts.get(this.tokenSymbol, this.chainSlug)?.messengerWrapper
+    if (!messengerWrapperContract) {
+      throw new Error(`Messenger wrapper contract not found for ${this.chainSlug}.${this.tokenSymbol}`)
+    }
+    this.messengerWrapper = new MessengerWrapper(messengerWrapperContract)
+
     // xDomain relayer is less time sensitive than others
     this.pollIntervalMs = 10 * 60 * 1000
   }
 
   async pollHandler () {
-    await this.checkTransfersCommittedFromDb()
+    await Promise.all([
```

Comment:
any consequences of having both promises canceled/failed if one them throws here in the Promise.all?

Reply:
Interesting. I think the bonder could get into a bad spot if a tx throws on one of those on each poll. Do you think it makes sense to `try...catch` and/or serialize these calls?

Reply:
I think try/catch would be good since they don't need to be serialized